### DISCUSSION TITLE
WiFi networks configuration

### DISCUSSION THREAD

**Discussion by nullcount** [nix-community/raspberry-pi-nix#60] [Q&A]
I'm trying to make SD card image for Pi Zero 2W.
The example/default.nix was edited to remove the line `eth0.useDHCP = true;`
The SD image was built, flashed, and booted on PiZero2w

However, I cannot find how to connect to WiFi.
`ip addr` shows interfaces `lo` and `wlan0` (status: DOWN)
I tried `ifconfig wlan0 up` but the `wlan0` still shows (status: DOWN)

I'd like to use networkmanager to set the SSID and psk for wireless networks in the configuration.
What files/snippets should I edit before building another SD card?

Maybe it would be helpful to include a note on WiFi in the README.

**Comment by tstat** [comment]
Hey @nullcount. I believe you are basically asking "How do I setup wifi on NixOS?". The relevant section of the NixOS manual is [here](https://nixos.org/manual/nixos/stable/#sec-networking).

You could configure it directly in the config [as demonstrated here](https://nixos.org/manual/nixos/stable/#sec-wireless), or enable network manager ([manual](https://nixos.org/manual/nixos/stable/#sec-networkmanager)) and connect with something like `nmtui`.
